packages/core/strapi/src/cli/commands/transfer/command.ts

```diff
@@ -45,14 +45,142 @@ const command = () => {
+            console.info(
+              '   → Add STRAPI_TRANSFER_URL and STRAPI_TRANSFER_TOKEN environment variables to make the transfer process faster for future runs'
+            );
+            return;
+          }
+
+          console.info('ℹ️  Found transfer configuration in your environment:');
+
+          if (hasEnvUrl) {
+            console.info(
+              `   → Environment STRAPI_TRANSFER_URL (${hasEnvUrl}) will be used as the transfer URL`
+            );
+          }
+
+          if (hasEnvToken) {
+            console.info(
+              '   → Environment STRAPI_TRANSFER_TOKEN value will be used as the transfer token'
+            );
+          }
+
+          console.info(); // Empty line for better readability
+        };
+
+        const determineDirection = async () => {
+          // If user has not provided a direction from CLI, log the documentation
+          if (!opts.from && !opts.to) {
+            logDocumentation();
+          }
+
+          logEnvironmentVariables();
+
+          if (opts.from) {
+            return opts.from;
+          }
+          if (opts.to) {
+            return opts.to;
+          }
+
+          const { dir } = await inquirer.prompt([
+            {
+              type: 'list',
+              name: 'dir',
+              message: 'Choose transfer direction:',
+              choices: [
+                { name: 'Pull data from remote Strapi to local', value: 'from' },
+                { name: 'Push local data to remote Strapi', value: 'to' },
+              ],
+            },
+          ]);
+
+          return dir;
+        };
+
+        const determineUrl = async (direction: 'from' | 'to') => {
+          if (opts[direction]) {
+            return new URL(opts[direction]);
+          }
+
+          if (process.env.STRAPI_TRANSFER_URL) {
+            return new URL(process.env.STRAPI_TRANSFER_URL);
+          }
+
+          const answer = await inquirer.prompt([
+            {
+              type: 'input',
+              name: 'remoteUrl',
+              message: `Enter the URL of the remote Strapi instance to ${direction === 'from' ? 'get data from' : 'send data to'}:`,
+              default: process.env.STRAPI_TRANSFER_URL,
+              validate(input: string) {
+                try {
+                  const url = new URL(input);
+                  if (!['http:', 'https:'].includes(url.protocol)) {
+                    return 'URL must use http: or https: protocol';
+                  }
+                  return true;
+                } catch (error) {
+                  return 'Please enter a valid URL (e.g., http://localhost:1337/admin or https://example.com/admin)';
+                }
+              },
+            },
+          ]);
+
+          return new URL(answer.remoteUrl);
+        };
+
+        const determineToken = async (direction: 'from' | 'to') => {
+          if (opts[`${direction}Token`]) {
```

Comment:
So if a token is provided, it takes precedence over the env value?

Reply:
Saw that afterward, but yeah https://github.com/strapi/strapi/pull/23421#issuecomment-2850485479

Reply:
Yeah, that's how I usually expect most tools to work. env values used by default, but provide a mechanism to override it case-by-case using the cli.

Reply:
lgtm this way